src/ssl.c

```diff
@@ -43856,88 +43843,286 @@ void* wolfSSL_SESSION_get_ex_data(const WOLFSSL_SESSION* session, int idx)
 }
 #endif /* OPENSSL_EXTRA || WOLFSSL_WPAS_SMALL || FORTRESS */
 
-/* Note: This is a huge section of API's - through
- *       wolfSSL_X509_OBJECT_get0_X509_CRL */
-#if defined(OPENSSL_ALL) || (defined(OPENSSL_EXTRA) && \
-    (defined(HAVE_STUNNEL) || defined(WOLFSSL_NGINX) || \
-    defined(HAVE_LIGHTY) || defined(WOLFSSL_HAPROXY) || \
-    defined(WOLFSSL_OPENSSH) || defined(HAVE_SBLIM_SFCB)))
-int wolfSSL_SESSION_get_ex_new_index(long idx, void* data, void* cb1,
-       void* cb2, CRYPTO_free_func* cb3)
+#if defined(OPENSSL_EXTRA) || defined(HAVE_STUNNEL) || defined(WOLFSSL_NGINX) || \
```

Comment:
I still don't really understand why this change is being made.

> Without these functions we are re-implementing a lot of the stack manipulation code.

The functions only exist once in ssl.c; what do you mean by this? I don't see how we're re-implementing anything.

Reply:
If you look in `internal.c` and `ssl.c` for this PR, you will find many instances where I refactored functions to use the `wolfSSL_sk_X509_NAME_*` API instead of directly manipulating the stack. Examples include `SSL_CtxResourceFree`, `wolfSSL_load_client_CA_file`, `wolfSSL_CTX_add_client_CA`. I believe that's all of them in this PR but there may be more.